### DISCUSSION TITLE
Variable font not working with Harfbuzz

### DISCUSSION THREAD

**Discussion by tulsiamin** [harfbuzz/harfbuzz#3881] [Q&A] [answered]
I am creating hb_font from freetype face by executing the following code.
```
hb_font_t* harfbuzzFontHandler = hb_ft_font_create(ft_face, nullptr);
```
 Now I am executing this piece of code just before running the shaping code.
```
 hb_face_t* hb_face = hb_font_get_face(harfbuzzFontHandler);
	if (hb_face && hb_ot_var_has_data(hb_face)) {
		const unsigned int variation_count = 2;
		hb_variation_t variation_data[variation_count];
		variation_data[0].tag = HB_OT_TAG_VAR_AXIS_WIDTH;
		variation_data[1].tag = HB_OT_TAG_VAR_AXIS_WEIGHT;
		variation_data[0].value = 99;
		variation_data[1].value = 750;
		hb_font_set_variations(harfbuzzFontHandler, variation_data, variation_count);
	}
```
It does not make the glyphs bold or does not increase the width.  What I suspect is that "hb_font_set_variations" internally works on hb_face and we have created hb_font based on FT_face which is why this does not work well. 
It would be nice to know the exact reason behind this or some corrections that would be needed.

**Comment by behdad** [comment, marked as answer]
Try calling `hb_ft_hb_font_changed()` after you changed the settings.

**Reply by tulsiamin** [reply]
@behdad 
Thank you for the reply, We are on version 2.6 and this api requires version 4.4, I will upgrade the version and let you know if it works but I had another very important question related to variable fonts.

When I use this function "hb_font_set_variations" , will it directly impact the values inside "hb_buffer_get_glyph_positions"?
hb_face_t* hb_face = hb_font_get_face(harfbuzzFontHandler);
	if (hb_face && hb_ot_var_has_data(hb_face)) {
		const unsigned int variation_count = 2;
		hb_variation_t variation_data[variation_count];
		variation_data[0].tag = HB_OT_TAG_VAR_AXIS_WIDTH;
		variation_data[1].tag = HB_OT_TAG_VAR_AXIS_WEIGHT;
		variation_data[0].value = 99;
		variation_data[1].value = 750;
		hb_font_set_variations(harfbuzzFontHandler, variation_data, variation_count);
	}
	//shape
	hb_shape(harfbuzzFontHandler, mTextBuffer, nullptr, 0);
	
       // Get glyphs data.
	unsigned int numGlyphs = 0;
	hb_glyph_info_t* glyphsInfo = hb_buffer_get_glyph_infos(mTextBuffer, &numGlyphs);
	hb_glyph_position_t* glyphsPositions = hb_buffer_get_glyph_positions(mTextBuffer, &numGlyphs);
	
Will the x_offset and y_offset change for the glyphs with and without usage of "hb_font_set_variations". We don't have a font which has x_offset and y_offset as non-zero values currently so we aren't able to figure that out.

**Reply by behdad** [reply]
> When I use this function "hb_font_set_variations" , will it directly impact the values inside "hb_buffer_get_glyph_positions"?

Yes, it does.
